### DISCUSSION TITLE
Why am I getting this error when I use the example code in the documentation?

### DISCUSSION THREAD

**Discussion by jelilat** [explosion/spaCy#7297] [Help: Coding & Implementations] [answered]
Invalid pattern specified: expected both SPEC and PATTERN.

Here is the code
`import spacy
from spacy.matcher import DependencyMatcher

nlp = spacy.load("en_core_web_sm")
matcher = DependencyMatcher(nlp.vocab)
pattern = [
  {
    "RIGHT_ID": "anchor_founded",       # unique name
    "RIGHT_ATTRS": {"ORTH": "founded"}  # token pattern for "founded"
  }
]
matcher.add("FOUNDED", [pattern])
doc = nlp("Smith founded two companies.")
matches = matcher(doc)
print(matches) # [(4851363122962674176, [1])]`

## Your Environment

- Operating System: Windows
- Python Version Used: Python 3
- spaCy Version Used: v3
- Environment Information: Kaggle Notebook

**Comment by adrianeboyd** [comment, marked as answer]
Your notebook environment must have spacy v2.x installed instead of spacy v3.x.

**Comment by svlandeg** [comment]
(Maintenance note: this question is well suited for the discussions forum, so I'm moving it there. While this issue gets closed, the discussion thread will remain open to continue the conversation if need be!)
